extern/sector-storage/piece_provider_test.go

```diff
@@ -0,0 +1,118 @@
+package sectorstorage
+
+import (
+	"context"
+	"io/ioutil"
+	"strings"
+	"testing"
+	"time"
+
+	"github.com/filecoin-project/go-state-types/abi"
+	"github.com/filecoin-project/go-statestore"
+	specstorage "github.com/filecoin-project/specs-storage/storage"
+	"github.com/ipfs/go-datastore"
+	"github.com/ipfs/go-datastore/namespace"
+	ds_sync "github.com/ipfs/go-datastore/sync"
+	"github.com/stretchr/testify/require"
+
+	"github.com/filecoin-project/lotus/extern/sector-storage/sealtasks"
+	"github.com/filecoin-project/lotus/extern/sector-storage/stores"
+	"github.com/filecoin-project/lotus/extern/sector-storage/storiface"
+)
+
+// TestPieceProviderReadPiece verifies that the ReadPiece method works correctly
+func TestPieceProviderReadPiece(t *testing.T) {
+	ctx, cancel := context.WithTimeout(context.Background(), 5*time.Minute)
+	defer cancel()
+
+	runTest := func(t *testing.T, alreadyUnsealed bool) {
+		// Set up sector storage manager
+		storage := newTestStorage(t)
+		index := stores.NewIndex()
+		localStore, err := stores.NewLocal(ctx, storage, index, nil)
+		require.NoError(t, err)
+		remoteStore := stores.NewRemote(localStore, index, nil, 6000)
+		dstore := ds_sync.MutexWrap(datastore.NewMapDatastore())
+		wsts := statestore.New(namespace.Wrap(dstore, datastore.NewKey("/worker/calls")))
+		smsts := statestore.New(namespace.Wrap(dstore, datastore.NewKey("/stmgr/calls")))
+		sealerCfg := SealerConfig{
+			ParallelFetchLimit: 10,
+			AllowAddPiece:      true,
+			AllowPreCommit1:    true,
+			AllowPreCommit2:    true,
+			AllowCommit:        true,
+			AllowUnseal:        true,
+		}
+		mgr, err := New(ctx, localStore, remoteStore, storage, index, sealerCfg, wsts, smsts)
+		require.NoError(t, err)
+
+		// Set up worker
+		localTasks := []sealtasks.TaskType{
+			sealtasks.TTAddPiece, sealtasks.TTPreCommit1, sealtasks.TTCommit1, sealtasks.TTFinalize, sealtasks.TTFetch,
+		}
+		testWorker := newTestWorker(WorkerConfig{
```

Comment:
@dirkmc 

We should try and use an actual worker here instead of a test worker. You can take a look at how we create an actual worker in one of the worker tests in manager_test.go

Reply:
@dirkmc I've added commit `3eded3df9dd6edf4e607c7af49026ed995790165` to this PR to use an actual worker here with the actual rust proofs library.

And the tests works fine ! Looking good here :)